### PR TITLE
ipmitool-xcat version 1.8.17 build tools

### PR DESCRIPTION
**Pull request by pdlun92**
Added ipmitool-1.8.17 and necessary patches.  Modified build scripts and spec file accordingly

Addresses xcat-core issue #871 (https://github.com/xcat2/xcat-core/issues/871)

### CODE REVIEW

**Comment by daniceexi** [comment]
@chenglch Please review this change.

File: ipmitool/ipmitool.spec [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
+Version:      1.8.17
+Release:      0
 License:      BSD
 Group:        Utilities
 Packager:     IBM Corp.
 Source:       ipmitool-%{version}.tar.gz
 Patch:        ipmitool-%{version}-saneretry.patch
 Patch2:       ipmitool-%{version}-rflash.patch
-Patch3:       ipmitool-%{version}-solactivate.patch
+#Patch3:       ipmitool-%{version}-solactivate.patch
```
Comment (whowutwut) [inline ipmitool/ipmitool.spec]: @pdlun92  I thought we had taken a look at this solactivate.patch and I thought we saw that it was included in 1.8.17 ,but now i'm looking at the source code and I don't see the following patch: 

https://github.com/xcat2/xcat-dep/blob/master/ipmitool/ipmitool-1.8.15-solactivate.patch

I don't see it in this file:  https://sourceforge.net/p/ipmitool/source/ci/IPMITOOL_1_8_17/tree/lib/ipmi_sol.c  

I think we need to create this new patch  `ipmitool-1.8.17-solactivate.patch` and rebuild the rpms.  :-( 

This is the thread that talks about it, but i thought I had seen a "target" for when this patch would be integrated,  I can't find it at this time.  https://sourceforge.net/p/ipmitool/mailman/message/34370504/

File: ipmitool/ipmitool.spec [hunk 2]
```diff
@@ -35,7 +35,7 @@ fi
 %setup -n ipmitool-%{version}
 %patch -p1
 %patch2 -p1
-%patch3 -p1
+#%patch3 -p1
```
Comment (whowutwut) [inline ipmitool/ipmitool.spec]: This would need uncomment if we are building the patch again.

**Review by whowutwut — CHANGES_REQUESTED** [review]

File: ipmitool/ipmitool.spec (hunk 1 above)
Comment (chenglch) [reply ipmitool/ipmitool.spec]: I'm sure that the code in the solactivate.patch is not included in the 1.8.17 version.

**Comment by chenglch** [comment]
To build the deb package, more change is needed in debian directory

**Review by chenglch — COMMENTED** [review]
To build the deb package, more change is needed in `debian` directory.

**Review by whowutwut — APPROVED** [review]

**Comment by pdlun92** [comment]
@chenglch thanks for the comments.  What platforms should we build the debian packages on?

**Comment by chenglch** [comment]
@pdlun92 , debian package is for ubuntu, if we do not support ubuntu system currently, it's ok

**Comment by whowutwut** [comment]
I saw a 1.8.15-3 .deb package. Which OS should be used to build it?

**Comment by chenglch** [comment]
Hi @whowutwut , ubuntu 14.04 system is used to build the deb package. We need some changes  and the symbol link files in the debian directory, then runs the `make_deb.sh` to build the deb package. We'd better try the deb package on ubuntu16.04 system to see if the package built on ubuntu14.04 woks well.

**Comment by chenglch** [comment]
@whowutwut , this comment is not about this pull request, but some thought about our procedure to change or build the dependency, especially for ipmitool which is changed a little frequently these days. Actually, generating the  diff-patch files, synchronizing the upstream version and building the packages on various platforms, all of them are hard work. We'd better find a new way to solve these problems in the future.

**Comment by zet809** [comment]
@whowutwut @pdlun92 Hi, victor and pat, I had noticed that the ipmitool-xcat-1.8.17 have been included in xcat-dep for Rh7, do you think we can ship it before doing a lot of verification with it? For example, rcons, rflash.

**Comment by whowutwut** [comment]
@zet809  understand your concern.  There is a requirement for Garrison to be at ipmitool 1.8.17 level and 2.12.3 is planned for ESP.   I think we should make sure we update all our environment to use 1.8.17 in the next few days to make sure we flush out any issues.  Official support for those servers will be 2.12.4 and/or 2.13 release.

**Comment by daniceexi** [comment]
If we need to use **ipmitool 1.8.17** for 2.12.3, please @tingtli and FVT team take some time to verify it for rh7.3 against FS and GRS server before the release manually.

**Comment by chenglch** [comment]
For rhels is OK, I agree to merge

**Pull request merged**